src/store/group-management/saga.ts

```diff
@@ -1,22 +1,47 @@
-import { SagaActionTypes, Stage, setStage } from '.';
-import { call, fork, put, race, take, delay } from 'redux-saga/effects';
+import { SagaActionTypes, Stage, setStage, setRoomMembers } from '.';
+import { Chat, chat } from '../../lib/chat';
+import { call, fork, put, race, take, select } from 'redux-saga/effects';
 import { Events, getAuthChannel } from '../authentication/channels';
+import { denormalize as denormalizeUsers } from '../users';
 
 export function* reset() {
+  yield put(setRoomMembers([]));
   yield put(setStage(Stage.None));
 }
 
+export function* roomMembersSelected(action) {
+  const { users: selectedMembers, roomId } = action.payload;
+
+  try {
+    if (!roomId || !selectedMembers) {
+      return;
+    }
+
+    const userIds = selectedMembers.map((user) => user.value);
+    const users = yield select((state) => denormalizeUsers(userIds, state));
+
+    const chatClient: Chat = yield call(chat.get);
+    yield call([chatClient, chatClient.addMembersToRoom], roomId, users);
+    yield put(setRoomMembers(selectedMembers));
+
+    return Stage.None;
+  } catch (error) {}
+}
+
 export function* saga() {
   yield fork(authWatcher);
 
   while (true) {
-    const { startEvent } = yield race({
+    const { startEvent, addSelectedMembersEvent } = yield race({
       startEvent: take(SagaActionTypes.StartAddMember),
+      addSelectedMembersEvent: take(SagaActionTypes.AddSelectedMembers),
```

Comment:
I'm confused why we're taking this event here AND on line 86. We should only ever get this event after we've started the process, right? So, we shouldn't need to listen for it here, I don't think.

Reply:
Yes correct, don't need to listen for it here. Thanks, updated.